Applications/Spire/Include/Spire/Ui/NumericInputWidget.hpp

```diff
@@ -0,0 +1,127 @@
+#ifndef SPIRE_NUMERIC_INPUT_WIDGET_HPP
+#define SPIRE_NUMERIC_INPUT_WIDGET_HPP
+#include <boost/multiprecision/cpp_dec_float.hpp>
+#include <boost/optional.hpp>
+#include <QAbstractSpinBox>
+#include <QLocale>
+#include <QStyle>
+#include <QRegularExpression>
+#include "Spire/Spire/Spire.hpp"
+
+namespace Spire {
+
+  //! Represents a widget for entering numeric values.
+  class NumericInputWidget : public QAbstractSpinBox {
+    public:
+
+      //! The maximum number of accepted decimal places.
+      static const auto MAX_DECIMAL_PLACES = 15;
```

Comment:
Using a value of ```15``` makes this no better than using a ```double``` which is not sufficient. Declare this ias ```constexpr``` if you intent to use it within a constant expression.

Reply:
Do you have a preferred max decimal count for this widget?